controller/setting_controller.go

```diff
@@ -323,11 +323,16 @@ func (sc *SettingController) syncDangerZoneSettingsForManagedComponents(settingN
 				return err
 			}
 		case types.SettingNameStorageNetwork:
-			fnAreAllVolumeDetached := func() (bool, error) {
-				isDetached, _, err := sc.ds.AreAllVolumesDetached(longhorn.DataEngineTypeAll)
-				return isDetached, err
+			detached, _, err := sc.ds.AreAllVolumesDetached(longhorn.DataEngineTypeAll)
+			if err != nil {
+				return errors.Wrapf(err, "failed to check volume detachment for %v setting update", types.SettingNameStorageNetwork)
+			}
+
+			if !detached {
+				return &types.ErrorInvalidState{Reason: fmt.Sprintf("failed to apply %v setting to Longhorn components when there are attached volumes. It will be eventually applied", types.SettingNameStorageNetwork)}
```

Comment:
NIT: It looks like we will log this error every time the setting rereconciles, even if no changes are required. The previous implementation (and the other similar ones) checked whether there was anything in need of updating BEFORE checking if volumes were detached.

Reply:
Good catch. Thank you.